### PR TITLE
RS.Backup New Features

### PR DESCRIPTION
**Pull request by dragonfly91**
## Description

1. Added -Vault parameter to the applicable cmdlets of RecoveryServices.Backup as an alternative to using Set-AzureRmRecoveryServicesVaultContext cmdlet. 
2. Added new cmdlet Get-AzureRmRecoveryServicesBackupStatus which helps in checking whether an Azure Resource is backed up by some Recovery Services Vault in the subscription.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by dragonfly91** [comment]
@vladimir-shcherbakov / @cormacpayne, I've fixed the merge conflicts. Please review the PR!

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    public class GetAzureRmRecoveryServicesBackupStatus : RecoveryServicesBackupCmdletBase
+    {
+        const string NameParamSet = "Name";
+        const string IdParamSet = "Id";
+
+        /// <summary>
+        /// Name of the Azure Resource whose representative item needs to be checked 
+        /// if it is already protected by some Recovery Services Vault in the subscription.
+        /// </summary>
+        [Parameter(ParameterSetName = NameParamSet,
```
Comment (cormacpayne) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs]: @dragonfly91 nit: can you put a `Mandatory = true` or `Mandatory = false` on each of the parameters?

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Helpers;
+using Microsoft.Azure.Commands.RecoveryServices.Backup.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
+{
+    /// <summary>
+    /// This cmdlet can be used to check if a VM is backed up by any vault in the subscription.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmRecoveryServicesBackupStatus")]
```
[the remark this answers is not in the document]
Comment (dragonfly91) [reply src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs]: @pvrk can you please comment on the cmdlet name?

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Commands.RecoveryServices.Backup.Providers.csproj [hunk 3]
```diff
@@ -73,6 +77,10 @@
       <Project>{3819d8a7-c62c-4c47-8ddd-0332d9ce1252}</Project>
       <Name>Commands.ResourceManager.Common</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\RecoveryServices\Commands.RecoveryServices\Commands.RecoveryServices.csproj">
```
[the remark this answers is not in the document]
Comment (dragonfly91) [reply src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Commands.RecoveryServices.Backup.Providers.csproj]: @cormacpayne we've expressed our concerns over mail. Can you please comment over there?

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        const string NameParamSet = "Name";
+        const string IdParamSet = "Id";
+
+        /// <summary>
+        /// Name of the Azure Resource whose representative item needs to be checked 
+        /// if it is already protected by some Recovery Services Vault in the subscription.
+        /// </summary>
+        [Parameter(ParameterSetName = NameParamSet,
+            HelpMessage = ParamHelpMsgs.ProtectionCheck.Name)]
+        public string Name { get; set; }
```
[the remark this answers is not in the document]
Comment (dragonfly91) [reply src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup/Cmdlets/Backup/GetAzureRmRecoveryServicesBackupStatus.cs]: Good catch! No, we need all the three - Name, ResourceGroupName and Type - to be given in the case of the NameParamSet. In the case of IdParamSet, only Id is expected from the user. How to enforce this?

**Comment by markcowl** [comment]
Per previous discussion, closing this review until consolidation and usage of common compute library

**Pull request closed without merging**